### DISCUSSION TITLE
Allow templating in output flag

### DISCUSSION THREAD

**Discussion by secustor** [lindell/multi-gitter#350] [Ideas]
Thanks for this very useful piece of code. It has been very helpful. 

I'm using it to gather data from multiple repos and then concat the result using the `print` sub command. 
The problem is that the resulting file is getting too big for efficient processing 

A solution would be to allow templating in the `output` flag that way such a use case would be possible: 
```shell
multi-gitter print report.sh -o myorg --output "./results/{{ .REPOSITORY }}.json"
```

Glancing at the code base this should be doable by moving the creation of the output stream to 
https://github.com/lindell/multi-gitter/blob/12c04221fa678bf230a50f4a4386aebe4cfa593f/internal/multigitter/print.go#L61

If you don't see blockers, I will look into implementing it.

**Comment by lindell** [comment]
I think the same thing could be archived adding the repo name by using $REPOSITORY and piping the output through a script that is splitting it. But I would absolutely accept a PR if you want to implement this within multi-gitter ☺️

**Reply by lindell** [reply]
This is specifically what I'm thinking about, since it seems that you output json, let's use that in the example.

`split.sh`:
```sh
#!/bin/bash

while read line
do
    repo=`echo $line | jq -r .repo` # Get the name of the repository from the top level json
    mkdir -p -- "${repo%/*}" # Make sure the folder exist
    echo $line >> $repo # Add to the file
done < /dev/stdin
```

And `test-script.sh`:
````sh
#!/bin/bash

echo "{\"repo\":\"$REPOSITORY\"}" # This is ofc more advanced in reality 
````

Then run with:
```sh
multi-gitter print ./test-script.sh -O my-cool-org | ./test.sh
```

Result:
![image](https://user-images.githubusercontent.com/284868/233710109-1f01fee3-0632-435e-9de3-012089b23101.png)

**Reply by secustor** [reply]
Thanks, for the example code! 

But I have started implementing this in code anyway, because it simplifying the scripting which would be necessary.    

It works, but I have still some regression. If I have fixed it up, I will provide a PR.
